library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/recorder/mapper/BaseWireframeMapper.kt

```diff
@@ -21,8 +20,11 @@ internal abstract class BaseWireframeMapper<T : View, S : MobileSegment.Wirefram
 
     protected fun colorAndAlphaAsStringHexa(color: Int, alphaAsHexa: Long): String {
         // we shift left 8 bits to make room for alpha
-        val colorAndAlphaAsHexa = (color.toLong().shl(8)).or(alphaAsHexa)
-        return String.format(Locale.US, "#%08X", 0xFFFFFFFF and colorAndAlphaAsHexa)
+        val colorAndAlpha = (color.toLong().shl(8)).or(alphaAsHexa)
+        val colorAndAlphaAsHexa = (0xffffffff and colorAndAlpha)
+            .toString(16)
+            .padStart(8, '0')
```

Comment:
Can you give some details on why this algorithm is more optimized vs the `String.format` one? Did you benchmark this?

Reply:
Yes, I profiled this with the Android Profiler...there is a big difference there.

Reply:
Because we are using this function for each tree node...I profiled the whole snapshot function. With the String.format the difference is ~10ms + which is huge.

Reply:
Maybe it'd be nice to add a comment there to make sure we remember why we use a three line code instead of a one liner ;)